### ISSUE TITLE
Wrong depression ID in outputted depressions.shp and level shapefiles

### ISSUE THREAD

**Issue by geotom** [giswqs/lidar#23]
* lidar version: 0.6.1 (Forced to use this one because of #21)
* Python version: 3.8
* Operating System: Ubuntu (Docker)

### Description

When delineating depressions I use the following
```
sink_path = ExtractSinks(
    './test/dem.tif',
    1,
    './output'
)
DelineateDepressions(
    sink_path,
    1,
    0.5,
    0.2,
    './output'
    True
)
```

This runs through and produces me the general sinks (`regions.shp`) and the nested depressions (`depressions.shp`, `depressions_info.csv`). I understand the relation between both and want to assign to each region all the associated depressions and their geometries by looping through all the regions (ID's) and assigning them every depression (getting info from the  `depressions_info.csv`). But I stumbled upon an issue where I cannot associate depressions by their IDs to the according poygon geometry, because the `depressions.shp` IDs are wrong. While I have enumerated all depressions correctly in the CSV file, the depressions shapefile has an very strange (and I assume wrong) numbering. At least most IDs fit, but I have for the level 1 depressions suddenly hundrets of depressions all having the ID 1. The rest of the polygons all have the correct depression ID, which I can also find in the CSV file.

![image](https://user-images.githubusercontent.com/5240363/161511189-ad8958aa-058a-416d-aa07-7e1909b1a5db.png)

The same can be sen when looking at the individual level shapefiles. For level 1, I have a lot of polygons all with the same depression ID = 1

![image](https://user-images.githubusercontent.com/5240363/161511886-8fae2b32-33f9-4c51-b6b5-34a9c8f289c8.png)

This does not allow to associate a geometry to a depression via its ID. I guess 1/50 or 1/100 of my depressions(and only those of level 1) have the wrong depression ID set to 1.

Do I misunderstand something or is this a bug?

**Comment by geotom** [comment]
So this would explain these artefacts? 
![image](https://user-images.githubusercontent.com/5240363/161640010-b75f68ef-5060-4592-a300-c2bd82878b9f.png)

I just checked the  `depression_id.if` which contains a lot of simple shaped features. All of the encircled raster areas have the `ID = 1`, and not the correct depression IDs.

**Comment by giswqs** [comment]
Level ID should be 1, but depression ID should not be 1. Make sure you look at the correct raster file, there is one named depression_id.tif and one named level_id.tif

**Comment by geotom** [comment]
Yes I am looking exactly at `depression_id.tif `. When I compare the values of this raster with the `depressions.shp` they exactly match. So I doubt it comes from a raster polygonisation process.

**Comment by geotom** [comment]
So what I can observe is that in the CSV file, all depressions get an unique ID, but in all geographical output (shapefiles, geotiff), the values are set to ID = 1 for a lot of depressions.
